internal/exec/stages/files/units.go

```diff
@@ -151,7 +152,16 @@ func (s *stage) createSystemdPresetFile(presets map[string]*Preset) error {
 		return err
 	}
 	hasInstanceUnit := false
-	for _, value := range presets {
+
+	// sort the units before writing to the systemd presets file
+	unitNames := make([]string, 0, len(presets))
+	for unit, _ := range(presets) {
+		unitNames = append(unitNames, unit)
+	}
+	sort.Strings(unitNames)
+
+	for i := range unitNames {
```

Comment:
Since we only need the values, you can do:

```suggestion
	for _, name := range unitNames {
```